### PR TITLE
Move addition of interactive log level to __init__

### PR DESCRIPTION
**Pull request by jsolbrig**
# Reviewer Checklist

PR author: Please ensure you meet all of the below requirements, and check boxes appropriately.

Reviewers: Please confirm all required testing/documentation has been completed prior to approving.

**Remove lines that are not applicable, explain if you select "NO REQUIRED"**

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] Required ***unit tests*** added and pass for new/modified functionality
* [x] NO REQUIRED ***integration tests*** No functionality actually changed
* [x] NO REQUIRED ***documentation*** No functionality actually changed
* [x] Required ***release notes*** added for new/modified functionality
* [x] NO REQUIRED ***updates to other repos*** No functionality actually changed

https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#NNN

# Testing Instructions

- [ ] Run base_test.sh - If this succeeds, the previous functionality still works.
- [ ] Ensure that the log.interactive calls work correctly, even when a specific plugin is accessed directly. For example, the following script used to raise an AttributeError:

```python
"""Test log.interactive in plugins."""

import os
from glob import glob
from geoips.interfaces import readers

data_path = (
    os.getenv("GEOIPS_TESTDATA_DIR") + "/test_data_noaa_aws/data/goes16/20200918/1950/"
)
data_files = glob(data_path + "*C1[45]*")

reader = readers.get_plugin("abi_netcdf")
data = reader(data_files)
```

# Summary

# Output

### CODE REVIEW

File: docs/source/releases/v1_12_2a0.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ | # # # Naval Research Laboratory, Marine Meteorology Division
+ | # # #
+ | # # # This program is free software: you can redistribute it and/or modify it under
+ | # # # the terms of the NRLMMD License included with this program. This program is
+ | # # # distributed WITHOUT ANY WARRANTY; without even the implied warranty of
+ | # # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+ | # # # for more details. If you did not receive the license, for more information see:
+ | # # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+
+Version 1.12.0 (2023-12-07)
```
Comment (evrose54) [inline docs/source/releases/v1_12_2a0.rst:13]: Need to update this line to ``Version 1.12.2a0 (2024-02-21)``

Comment (evrose54) [reply docs/source/releases/v1_12_2a0.rst:13]: And adjust the underline below it to match that new string length

**Comment by evrose54** [comment]
I ran both base_test.sh and the script you provided and they both passed. Once the comments above are addressed, we should be good to move forward on this one.

File: docs/source/releases/v1_12_2a0.rst (hunk 1 above)
Comment (jsolbrig) [reply docs/source/releases/v1_12_2a0.rst]: Thanks Evan.

File: docs/source/releases/v1_12_2a0.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    modified: geoips/__init__.py
+    modified: geoips/commandline/log_setup.py
+    added: tests/unit_tests/commandline/log_setup.py
+
+
+Refactor
+========
+
+Make JSON Plugin Registries Readable
```
Comment (mindyls) [inline docs/source/releases/v1_12_2a0.rst:45]: Should this be in the bulleted list above?

Comment (jsolbrig) [reply docs/source/releases/v1_12_2a0.rst:46]: Fixed, thanks!

File: docs/source/releases/v1_12_2a0.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ | # # # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the included license
+ | # # # for more details. If you did not receive the license, for more information see:
+ | # # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
+
+Version 1.12.2a0 (2024-02-21)
+*****************************
+
+* Bug fixes
+
+  * Ensure that log.interactive works from plugins when imported independently
```
Comment (mindyls) [inline docs/source/releases/v1_12_2a0.rst:18]: Isn't this doing a lot more than moving the interactive log level to __init__?  Did you update the whole log_setup to be class-based within this PR?

Comment (jsolbrig) [reply docs/source/releases/v1_12_2a0.rst:18]: Yes, yes I did. I'll update this.

Comment (jsolbrig) [reply docs/source/releases/v1_12_2a0.rst]: Updated.

File: docs/source/releases/v1_12_2a0.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+* Refactor
+
+  * Make JSON Plugin Registries Readable
+
+
+
+Bug Fixes
+=========
+
+Make add_log_level a callable class, ensure log.interactive works consistently
```
Comment (mindyls) [inline docs/source/releases/v1_12_2a0.rst:30]: I added this header @jsolbrig

File: tests/unit_tests/commandline/log_setup.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    assert "FROM PYTEST" in caplog.text
+    assert "INTERACTIVE" in caplog.text
+
+
+def test_log_interactive_from_directly_imported_plugin(caplog):
+    """Use ABI reader, which calls log.interactive.
+
+    It used to be that plugins loaded via get_plugin outside of GeoIPS could not call
+    log.interactive. They would raise an AttributeError. This is to ensure that problem
+    has been fixed and remains fixed.
```
Comment (mindyls) [inline tests/unit_tests/commandline/log_setup.py:37]: hooray

**Review by mindyls — APPROVED** [review]
I did not test myself, but assuming base_test.sh passed, and importing from readers directly worked (you could leave a screen shot or output snippet in the main conversation), I am good to go with this!

File: docs/source/releases/v1_12_2a0.rst (hunk 4 above)
Comment (jsolbrig) [reply docs/source/releases/v1_12_2a0.rst:30]: Thanks Mindy.

**Review by evrose54 — APPROVED** [review]
LGTM

**Pull request merged**